go/libraries/doltcore/merge/merge_schema_test.go

```diff
@@ -0,0 +1,585 @@
+			{commands.AddCmd{}, []string{"."}},
+			{commands.CommitCmd{}, []string{"-m", "modified branch master"}},
+			{commands.CheckoutCmd{}, []string{"other"}},
+			{commands.SqlCmd{}, []string{"-q", "alter table test modify c2 int not null;"}},
+			{commands.AddCmd{}, []string{"."}},
+			{commands.CommitCmd{}, []string{"-m", "modified branch other"}},
+			{commands.CheckoutCmd{}, []string{"master"}},
+		},
+		sch: schemaFromColsAndIdxs(
+			colCollection(
+				newColTypeInfo("pk", uint64(0), typeinfo.Int32Type, true, schema.NotNullConstraint{}),
+				newColTypeInfo("c1", uint64(1), typeinfo.Int32Type, false),
+				newColTypeInfo("c2", uint64(2), typeinfo.Int32Type, false, schema.NotNullConstraint{}),
+				newColTypeInfo("c3", uint64(3), typeinfo.Int32Type, false)),
+			schema.NewIndex("c1_idx", []uint64{1}, []uint64{1, 0}, nil, schema.IndexProperties{}),
+		),
+	},
+	{
+		name: "add index, drop index, merge",
+		setup: []testCommand{
+			{commands.SqlCmd{}, []string{"-q", "create index c3_idx on test(c3);"}},
+			{commands.AddCmd{}, []string{"."}},
+			{commands.CommitCmd{}, []string{"-m", "modified branch master"}},
+			{commands.CheckoutCmd{}, []string{"other"}},
+			{commands.SqlCmd{}, []string{"-q", "alter table test drop index c1_idx;"}},
+			{commands.AddCmd{}, []string{"."}},
+			{commands.CommitCmd{}, []string{"-m", "modified branch other"}},
+			{commands.CheckoutCmd{}, []string{"master"}},
+		},
+		sch: schemaFromColsAndIdxs(
+			colCollection(
+				newColTypeInfo("pk", uint64(0), typeinfo.Int32Type, true, schema.NotNullConstraint{}),
+				newColTypeInfo("c1", uint64(1), typeinfo.Int32Type, false, schema.NotNullConstraint{}),
+				newColTypeInfo("c2", uint64(2), typeinfo.Int32Type, false),
+				newColTypeInfo("c3", uint64(3), typeinfo.Int32Type, false)),
+			schema.NewIndex("c3_idx", []uint64{3}, []uint64{3, 0}, nil, schema.IndexProperties{}),
+		),
+	},
+	{
+		name: "rename columns",
+		setup: []testCommand{
+			{commands.SqlCmd{}, []string{"-q", "alter table test rename column c3 to c33;"}},
+			{commands.AddCmd{}, []string{"."}},
+			{commands.CommitCmd{}, []string{"-m", "modified branch master"}},
+			{commands.CheckoutCmd{}, []string{"other"}},
+			{commands.SqlCmd{}, []string{"-q", "alter table test rename column c2 to c22;"}},
+			{commands.AddCmd{}, []string{"."}},
+			{commands.CommitCmd{}, []string{"-m", "modified branch other"}},
+			{commands.CheckoutCmd{}, []string{"master"}},
+		},
+		sch: schemaFromColsAndIdxs(
+			colCollection(
+				newColTypeInfo("pk", uint64(0), typeinfo.Int32Type, true, schema.NotNullConstraint{}),
+				newColTypeInfo("c1", uint64(1), typeinfo.Int32Type, false, schema.NotNullConstraint{}),
+				newColTypeInfo("c22", uint64(2), typeinfo.Int32Type, false),
+				newColTypeInfo("c33", uint64(3), typeinfo.Int32Type, false)),
+			schema.NewIndex("c1_idx", []uint64{1}, []uint64{1, 0}, nil, schema.IndexProperties{}),
+		),
+	},
+	{
+		name: "rename indexes",
```

Comment:
There's also `ALTER TABLE test RENAME INDEX c1_idx TO c1_index` that you could add to cover another base.